runconfig/hostconfig.go

```diff
@@ -217,6 +218,17 @@ func MergeConfigs(config *Config, hostConfig *HostConfig) *ContainerConfigWrappe
 	}
 }
 
+func ValidateHostConfig(hostConfig *HostConfig) error {
```

Comment:
+1. I definitely felt that this was a little `hacky` when I wrote the verification. I couldn't think of a better place to put it though.